src/dsp/ethereum/fill_blocks.ts

```diff
@@ -59,37 +77,11 @@ export function fillInputOneBlock(input: any, zkgraphYaml: ZkGraphYaml, blockPre
         stateDSSlotsList[i],
         '\n',
       )
-
-      input.addHexString(stateDSAddrList[i], false) // address
-      // let ethproof = await getProof(
-      //   provider,
-      //   stateDSAddrList[i],
-      //   stateDSSlotsList[i],
-      //   ethers.utils.hexValue(blockNumber)
-      // );
-
-      const acctPrep = blockPrep?.getAccount(stateDSAddrList[i])
-
-      input.addVarLenHexString(acctPrep?.rlpNode, false) // account rlp
-      input.addVarLenHexStringArray(acctPrep?.accountProof, false) // account proof
-
-      const sourceSlots = stateDSSlotsList[i]
-      input.addInt(sourceSlots.length, false) // slot count
-
-      for (let j = 0; j < sourceSlots.length; j++) {
-        const slotPrep = acctPrep?.getSlot(sourceSlots[j])
-        // slot might doesn't exist. can't proceed in this case.
-        if (slotPrep.storageProof == null)
-          throw new Error(`In ExecInputGen: slot ${sourceSlots[j]} doesn't exist on given block height, storage proof == null. \n Please update yaml or use later blocknumber.`)
-
-        input.addHexString(sourceSlots[j], false)
-        input.addVarLenHexString(slotPrep.value, false)
-        input.addVarLenHexStringArray(slotPrep.storageProof, false)
-      }
     }
+    fillInputStorageFunc(input, blockPrep, stateDSAddrList, stateDSSlotsList)
   }
   else {
-    console.log('[*] No storage DS provided, skip...')
+    console.log('[*] No storage DS provided, skip...') // can rm
```

Comment:
Should we consider defining logs that are not easily migratable as an Error class?

e.g.
```js
class PainlessError extends Error {}
```

Reply:
nice try, but isn't it that it will break the execution flow?

Reply:
for logs that is not time sensitive, we can move them to the return object.
for time sensitive logs, probably a env defined hook console object is the best.
e.g. by default, use console; callers can pass in myConsole object, NOT through the func params but some env variables (because some func have logs, some not, not friendly)

Reply:
we can do this in the next version